profile/c5g7-cmfd.cpp

```diff
@@ -0,0 +1,509 @@
+  guide_tube_cell.addSurface(-1, &fuel_radius);
+
+  Universe guide_tube;
+  guide_tube.addCell(&guide_tube_cell);
+  guide_tube.addCell(&moderator_ring1);
+  guide_tube.addCell(&moderator_ring2);
+  guide_tube.addCell(&moderator_ring3);
+
+  /* Reflector */
+  Cell reflector_cell;
+  reflector_cell.setFill(materials["Water"]);
+
+  Universe reflector;
+  reflector.addCell(&reflector_cell);
+
+  /* Cells */
+  Cell assembly1_cell;
+  Cell assembly2_cell;
+  Cell refined_reflector_cell;
+  Cell right_reflector_cell;
+  Cell corner_reflector_cell;
+  Cell bottom_reflector_cell;
+
+  Universe assembly1;
+  Universe assembly2;
+  Universe refined_reflector;
+  Universe right_reflector;
+  Universe corner_reflector;
+  Universe bottom_reflector;
+
+  assembly1.addCell(&assembly1_cell);
+  assembly2.addCell(&assembly2_cell);
+  refined_reflector.addCell(&refined_reflector_cell);
+  right_reflector.addCell(&right_reflector_cell);
+  corner_reflector.addCell(&corner_reflector_cell);
+  bottom_reflector.addCell(&bottom_reflector_cell);
+
+  /* Root Cell */
+  Cell root_cell;
+  root_cell.addSurface(+1, &left);
+  root_cell.addSurface(-1, &right);
+  root_cell.addSurface(-1, &top);
+  root_cell.addSurface(+1, &bottom);
+
+  Universe root_universe;
+  root_universe.addCell(&root_cell);
+
+  /* Create lattices */
+  log_printf(NORMAL, "Creating lattices...");
+
+  /* Top left, bottom right 17 x 17 assemblies */
+  Lattice assembly1_lattice;
+  assembly1_lattice.setWidth(1.26, 1.26);
+  Universe* matrix1[17*17];
+  {
+    int mold[17*17] =  {1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1,
+                        1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 2, 1, 1, 2, 1, 1, 3, 1, 1, 2, 1, 1, 2, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1,
+                        1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+                        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1};
+
+    std::map<int, Universe*> names = {{1, &uo2}, {2, &guide_tube}, 
+                                      {3, &fission_chamber}};
+    for(int n=0; n<17*17; n++)
```

Comment:
This certainly works, or you could consider getting C++ fancy with [``std::transform``](http://en.cppreference.com/w/cpp/algorithm/transform). I honestly think we should consider making a ``Lattice`` utility method to assist with this templating concept, but that can be kept for a future PR.